lib/licenses.nix

```diff
@@ -230,6 +230,13 @@ in mkLicense lset) ({
     free = false;
   };
 
+  cc-by-nc-nd-40 = {
+    spdxIt = "CC-BY-NC-ND-4.0";
+    fullName = "Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International";
+    url = "https://spdx.org/licenses/CC-BY-NC-ND-4.0.html";
```

Comment:
```suggestion
    spdxId = "CC-BY-NC-ND-4.0";
    fullName = "Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International";
```

If you want to add software using this license, you can just include this commit in the PR that adds the software (but order it before the commit that uses the license).

Reply:
About the URL: Without specifying it manually, there is no `url` attribute.
![image](https://user-images.githubusercontent.com/12484955/232119581-30ee2155-84ce-4596-92f0-2706ec4bf256.png)

The CC-BY-NC-ND-3.0 is indeed auto-generated, but not for 4.0

Reply:
About the snippet and the commit: I am working on a personal project, packaging Herb Sutter's [cppfront project](https://github.com/hsutter/cppfront/) which is licensed as CC-BY-NC-ND-4.0. `cppfront` is nowhere near production ready to be included in Nixpkgs.

My current package [just write the license directly in-derivation](https://github.com/COLAMAroro/nixos-packages/blob/main/custom-packages/cppfront/default.nix#L7), but it would make it prettier if it was relying on a nixpkgs license, and that's something less for the next contributor trying to add a CC-BY-NC-ND-4.0 package :)

Reply:
there is no `url` attribute because you didn't specify `spdxId`.

Reply:
Oh crap that's a typo, nice catch, sorry I didn't catch that